src/rtp/rtcp/twcc.rs

```diff
@@ -1097,6 +1098,53 @@ impl TwccSendRegister {
         Some(&self.queue[index])
     }
 
+    /// Calculate the egress loss for given time window.
+    ///
+    /// **Note:** The register only keeps a limited number of records and using `duration` values
+    /// larger than ~1-2 seconds is liable to be inaccurate since some packets sent might have already
+    /// been evicted from the register.
+    ///
+    /// Packets sent very recently, within one RTT, are not considered because it's not likely that
+    /// a TWCC report could've been generated by the sender and received yet.
+    pub fn loss(&self, duration: Duration, now: Instant) -> Option<f64> {
+        let recent_rtt = self
+            .queue
+            .iter()
+            .rev()
+            .find_map(|r| r.rtt())
+            .map(|rtt| rtt.clamp(Duration::from_millis(5), Duration::from_millis(150)))
+            .unwrap_or(Duration::from_millis(50));
+
+        // Consider only packets that were sent at least one RTT ago. For more recent sends we will
+        // likely not have received the TWCC report in any case.
+        let upper_bound = now - recent_rtt;
+        // Consider only packets in the span specified by the caller
+        let lower_bound = now - duration;
+
+        let packets = self
+            .queue
+            .iter()
+            .rev()
+            .skip_while(|s| s.local_send_time >= upper_bound)
+            .take_while(|s| s.local_send_time >= lower_bound);
+
+        let (total, lost) = packets.fold((0, 0), |(total, lost), s| {
+            let was_lost = s
+                .recv_report
+                .as_ref()
+                .map(|rr| rr.remote_recv_time.is_none())
+                .unwrap_or(true);
+
+            (total + 1, lost + u64::from(was_lost))
+        });
+
+        if total == 0 {
+            return None;
+        }
+
+        Some((lost as f64) / (total as f64))
+    }
+
```

Comment:
This looks a bit expensive. Is it?

Reply:
Shouldn't be, it's `O(2n)` at worst with `n == 1000` currently and runs every second. I don't really think the `2n` case will happen either